specs/commandline/blob.md

```diff
@@ -0,0 +1,365 @@
+
+# Add a default signing key referencing the remote key identifier, and the plugin associated with it.
+notation key add --default --name <key_name> --plugin <plugin_name> --id <remote_key_id>
+
+# sign a blob
+notation blob sign --signature-name my-blob-signature /tmp/my-blob.bin
+```
+
+An example for a successful signing:
+
+```console
+$ notation blob sign --signature-name my-blob-signature /tmp/my-blob.bin
+Successfully signed /tmp/my-blob.bin
+Signature written to ./my-blob-signature.sig.jws
+```
+
+### Sign a blob with on-demand remote key
+
+```shell
+notation blob sign --plugin <plugin_name> --id <remote_key_id> --signature-name my-blob-signature /tmp/my-blob.bin
+```
+
+### Sign a blob using COSE signature format
+
+```shell
+# Prerequisites:
+# A default signing key is configured using CLI "notation key"
+
+# Use option "--signature-format" to set the signature format to COSE.
+notation blob sign --signature-format cose --signature-name my-blob-signature /tmp/my-blob.bin
+Successfully signed /tmp/my-blob.bin
+Signature written to ./my-blob-signature.sig.cose
+```
+
+### Sign a blob using the default signing key
+
+```shell
+# Prerequisites:
+# A default signing key is configured using CLI "notation key"
+
+notation blob sign --signature-name my-blob-signature /tmp/my-blob.bin
+```
+
+### Sign a blob with user metadata
+
+```shell
+# Prerequisites:
+# A default signing key is configured using CLI "notation key"
+
+# sign a blob and add user-metadata io.wabbit-networks.buildId=123 to the payload
+notation blob sign --user-metadata io.wabbit-networks.buildId=123 --signature-name my-blob-signature /tmp/my-blob.bin
+
+# sign a blob and add user-metadata io.wabbit-networks.buildId=123 and io.wabbit-networks.buildTime=1672944615 to the payload
+notation blob sign --user-metadata io.wabbit-networks.buildId=123 --user-metadata io.wabbit-networks.buildTime=1672944615 --signature-name my-blob-signature /tmp/my-blob.bin
+```
+
+### Sign a blob and specify the media type for the blob
+
+```shell
+notation blob sign --media-type <media type> --signature-name my-blob-signature /tmp/my-blob.bin
+```
+
+### Sign a blob and specify the signature expiry duration, for example 24 hours
+
+```shell
+notation blob sign --expiry 24h --signature-name my-blob-signature /tmp/my-blob.bin
+```
+
+### Sign a blob using a specified signing key
+
+```shell
+# List signing keys to get the key name
+notation key list
+
+# Sign a container image using the specified key name
+notation blob sign --key <key_name> --signature-name my-blob-signature /tmp/my-blob.bin
+```
+
+## Inspect detached blob signatures
+
+### Display details of the given detached blob signature and its associated certificate properties
+
+
+```text
+notation blob inspect [flags] /tmp/my-blob-signature.sig.jws
+```
+
+### Inspect the given detached blob signature
+
+```shell
+# Prerequisites: Signatures is produced by notation blob sign command
+notation blob inspect /tmp/my-blob-signature.sig.jws
```

Comment:
Can you add an example of the following command: `notation blob inspect /tmp/my-blob-signature.sig`, where `my-blob-signature.sig` is a valid signature following the Notation spec.

Reply:
This is what we have at the moment, correct? Can you please clarify how the command you provided is different from the existing one?

If you are referring to verifying a signature without signature format extension, I do not think that is something worth calling out in the docs. Notation should recommend users to keep extensions. Notation supporting no-extension signature files is only a convenience but should not be encouraged as it increases verification latencies.